R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
Right, I think that could go either way
```
scala> val a = SparkSession.builder().master("yarn").config("spark.master", "local[3]").getOrCreate()
scala> a.conf.get("master")
java.util.NoSuchElementException: master
scala> a.conf.get("spark.master")
res8: String = local[3]
```

I think the current R implementation mimic the programmatic behavior rather than commandline.

We could certainly change it - just want to highlight this is a breaking change if we are to silently flip the precedence order, and would think we should avoid unless if we have reason to.